### PR TITLE
Run tests locally with tox to create parity with CI. pep8 as part of tests.

### PR DESCRIPTION
**Pull request by mattiskan**
Makes make test run tox, the same way our CI does. Hopefully this makes for slightly faster iterations, and could help expose issues with different python versions. 

Also adding stricter requirements in terms of test coverage and code style.

I also think this new error message is more informative for missing `.env`:
```bash
bash-3.2$ make test
flake8 plaid
./.env tox
make: ./.env: No such file or directory
make: *** [test] Error 1
```

### CODE REVIEW

File: tests/integration/test_income.py [hunk 1]
```diff
@@ -1,8 +1,7 @@
 from plaid.errors import ItemError
 from tests.integration.util import (
     create_client,
-    CREDENTIALS,
-    SANDBOX_INSTITUTION
```
Comment (mattiskan) [inline tests/integration/test_income.py]: pep8 warned because they were unused.

File: tests/integration/test_processor.py [hunk 2]
```diff
@@ -1,9 +1,6 @@
 '''
 Client.Processor.* tests.
 '''
-
-from contextlib import contextmanager
-
```
Comment (mattiskan) [inline tests/integration/test_processor.py]: unused

File: tox.ini [hunk 3]
```diff
@@ -6,4 +6,6 @@ passenv = *
 deps=
   -rrequirements.txt
 commands=
-  py.test --cov plaid -s -rxs ./tests/
+  coverage run --source=plaid/,tests/ -m pytest --strict {posargs}
+  coverage report -m --show-missing --fail-under 96
+  flake8 .
```
Comment (mattiskan) [inline tox.ini]: This sets a hard lower limit on test coverage and fails on pep8 issues.
 I think it's good practice, but please confirm this is what we want.

**Comment by dsfish** [comment]
Thanks for taking this on! 😄

**Pull request merged**